vcd/resource_vcd_org_vdc_test.go

```diff
@@ -0,0 +1,196 @@
+// +build vdc ALL functional
+
+package vcd
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+var TestAccVcdVdc = "TestAccVcdVdcBasic"
+
+func TestAccVcdVdcBasic(t *testing.T) {
+
+	if !usingSysAdmin() {
+		t.Skip("TestAccVcdVdcBasic requires system admin privileges")
+		return
+	}
+
+	if testConfig.VCD.ProviderVdc.Name == "" {
+		t.Skip("Variable networking.Name must be set to run VDC tests")
+		return
+	}
+
+	if testConfig.VCD.ProviderVdc.NetworkPool == "" {
+		t.Skip("Variable networking.NetworkPool must be set to run VDC tests")
+		return
+	}
+
+	if testConfig.VCD.ProviderVdc.StorageProfile == "" {
+		t.Skip("Variable networking.StorageProfile must be set to run VDC tests")
+		return
+	}
+
+	var vdc govcd.Vdc
+	var params = StringMap{
+		"VdcName":                   TestAccVcdVdc,
+		"OrgName":                   testConfig.VCD.Org,
+		"AllocationModel":           testConfig.VCD.AllocationModel,
```

Comment:
We need to check that allocation model was set in the configuration file.

Reply:
allocation model removed, test improved to cover all cases